src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/OrderingHelper.cs

```diff
@@ -0,0 +1,320 @@
+    {
+        /// <summary>
+        /// Checks if the given project tree can move up over one of its siblings.
+        /// </summary>
+        public static bool CanMoveUp(IProjectTree projectTree)
+        {
+            Requires.NotNull(projectTree, nameof(projectTree));
+
+            return GetSiblingByMoveAction(projectTree, MoveAction.Up) != null;
+        }
+
+        /// <summary>
+        /// Move the project tree up over one of its siblings.
+        /// </summary>
+        public static bool MoveUp(UnconfiguredProject unconfiguredProject, Project project, IProjectTree projectTree)
+        {
+            Requires.NotNull(unconfiguredProject, nameof(unconfiguredProject));
+            Requires.NotNull(project, nameof(project));
+            Requires.NotNull(projectTree, nameof(projectTree));
+
+            return Move(unconfiguredProject, project, projectTree, MoveAction.Up);
+        }
+
+        /// <summary>
+        /// Move the project tree up over one of its siblings.
+        /// </summary>
+        public static Task<bool> MoveUpAsync(ConfiguredProject configuredProject, IProjectTree projectTree)
+        {
+            Requires.NotNull(configuredProject, nameof(configuredProject));
+            Requires.NotNull(projectTree, nameof(projectTree));
+
+            return MoveAsync(configuredProject, projectTree, MoveAction.Up);
+        }
+
+        /// <summary>
+        /// Checks if the given project tree can move down over one of its siblings.
+        /// </summary>
+        public static bool CanMoveDown(IProjectTree projectTree)
+        {
+            Requires.NotNull(projectTree, nameof(projectTree));
+
+            return GetSiblingByMoveAction(projectTree, MoveAction.Down) != null;
+        }
+
+        /// <summary>
+        /// Move the project tree down over one of its siblings.
+        /// </summary>
+        public static bool MoveDown(UnconfiguredProject unconfiguredProject, Project project, IProjectTree projectTree)
+        {
+            Requires.NotNull(unconfiguredProject, nameof(unconfiguredProject));
+            Requires.NotNull(project, nameof(project));
+            Requires.NotNull(projectTree, nameof(projectTree));
+
+            return Move(unconfiguredProject, project, projectTree, MoveAction.Down);
+        }
+
+        /// <summary>
+        /// Move the project tree down over one of its siblings.
+        /// </summary>
+        public static Task<bool> MoveDownAsync(ConfiguredProject configuredProject, IProjectTree projectTree)
+        {
+            Requires.NotNull(configuredProject, nameof(configuredProject));
+            Requires.NotNull(projectTree, nameof(projectTree));
+
+            return MoveAsync(configuredProject, projectTree, MoveAction.Down);
+        }
+
+        private enum MoveAction { Up=0, Down=1 }
+
+        private static void ProcessProjectTree(Queue<IProjectTree> treeQueue, HashSet<string> lookup, UnconfiguredProject unconfiguredProject, IProjectTree projectTree)
```

Comment:
I don't like that this method modifies both `treeQueue` and `lookup` but returns `void`. Perhaps return a tuple so its clear what the inputs and outputs are?

Reply:
I'll see what I can do.

Reply:
Done.